### ISSUE TITLE
Add support for export pre-process scripts

### ISSUE THREAD

**Issue by cloewen8** [godotengine/godot-proposals#2637]
### Describe the project you are working on
An erotic quick-time horror game that uses a CanvasModulate node to fade in and out scenes.

### Describe the problem or limitation you are having in your project
I want scenes to start faded out in exports, without them being faded out while editing. In more general terms, I want properties to be different in the exported game vs. while editing.

If scenes don't start faded out, then the scene will flash briefly before fading in.

### Describe the feature/enhancement and how it helps to overcome the problem or limitation
I think the best option would be to have something similar to `EditorScenePostImport`, where each scene is passed to a function that can manipulate nodes before it is compiled into the export.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
Here is what that could look like:

```gd
tool
extends EditorProjectPreExport

func pre_export_scene(scene): # Passed each scene in the project.
    # Do your stuff here.
    return scene # Return the modified scene (or perhaps nil to exclude it)

func pre_export_resource(res): # Passed all resources
    # Do your stuff here.
    return res # Return the modified resource (or nil to exclude)
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I couldn't find any documented way to hook into the exporting process. This doesn't stop me from doing anything and is rather a nice-to-have feature.

### Is there a reason why this should be core and not an add-on in the asset library?
There needs to be some way to hook into the exporting process, that needs to be core. I'd be happy with the remainder being an add-on!

**Comment by cloewen8** [comment]
> There is EditorExportPlugin. But you most likely don't need it, you can use `if OS.has_feature("standalone")` for code that runs only in the exported project.

Better than nothing! As far as I know, using a script, each node would need its own script with `_init` set (to set the properties before the nodes render). This seems like a lot of additional resources and work for such a small task.

I'll have to think about it. Currently, I feel like having an add-on to specify "export-only" properties would be the better option (but I could be thinking too lazy).

**Comment by KoBeWi** [comment]
No, you really don't need any pre-processing here. Check this project (try exporting it):
[FadeTest.zip](https://github.com/godotengine/godot-proposals/files/6364550/FadeTest.zip)
If each your game scene will inherit from a common class that does the fading, you need to write it only once and then it will require 0 additional code or anything.

**Comment by cloewen8** [comment]
> 
> 
> No, you really don't need any pre-processing here. Check this project (try exporting it):
> [FadeTest.zip](https://github.com/godotengine/godot-proposals/files/6364550/FadeTest.zip)
> If each your game scene will inherit from a common class that does the fading, you need to write it only once and then it will require 0 additional code or anything.

I completely forgot about Autoplay on Load. That fixes the flickering, which was my main reason for the pre-processing. My problem is solved, this proposal is no longer needed. Thank you!
